languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -150,17 +193,27 @@ def load_str(str, filename: nil) # rubocop:disable Metrics/AbcSize, Metrics/Meth
       #   @param query [Predicate]
       #   @return [Enumerator] of resulting bindings
       #   @raise [Error] if the FFI call raises one.
-      def query(query)
-        new_host = host.dup
+      def query(query, host: self.host.dup, bindings: {})
         case query
         when String
           ffi_query = ffi_polar.new_query_from_str(query)
         when Predicate
-          ffi_query = ffi_polar.new_query_from_term(new_host.to_polar(query))
+          ffi_query = ffi_polar.new_query_from_term(host.to_polar(query))
         else
           raise InvalidQueryTypeError
         end
-        Query.new(ffi_query, host: new_host)
+        Query.new(ffi_query, host: host, bindings: bindings)
+      end
+
+      # Query the knowledge base to determine whether an actor is allowed to
+      # perform an action upon a resource.
+      #
+      # @param actor [Object] Subject.
+      # @param action [Object] Verb.
+      # @param resource [Object] Object.
+      # @return [Boolean] An access control decision.
+      def allowed?(actor:, action:, resource:)
```

Comment:
Why did this get moved to polar.rb?

Reply:
to make it accessible from tests without changing them to use an `Oso` instead of a `Polar`